src/main/java/befaster/SendCommandToServer.java

```diff
@@ -53,14 +54,24 @@
      *
      **/
     public static void main(String[] args) throws ConfigNotFoundException {
-        ClientRunner.forUsername(readFromConfigFile("tdl_username"))
-                .withServerHostname(readFromConfigFile("tdl_hostname"))
-                .withActionIfNoArgs(RunnerAction.testConnectivity)
+
+        QueueBasedImplementationRunner runner = new QueueBasedImplementationRunner.Builder()
+                .setUniqueId(readFromConfigFile("tdl_username"))
+                .setHostname(readFromConfigFile("tdl_hostname"))
                 .withSolutionFor("sum", p -> Sum.sum(asInt(p[0]), asInt(p[1])))
                 .withSolutionFor("hello", p -> Hello.hello(p[0]))
                 .withSolutionFor("fizz_buzz", p -> FizzBuzz.fizzBuzz(asInt(p[0])))
                 .withSolutionFor("checkout", p -> Checkout.checkout(p[0]))
-                .start(args);
+                .create();
+
+        ChallengeSession.forUsername(readFromConfigFile("tdl_username"))
+                .withServerHostname(readFromConfigFile("tdl_hostname"))
+                .withColours(Boolean.parseBoolean(readFromConfigFile("tdl_use_coloured_output", "true")))
+                .withRecordingSystemOn(Boolean.parseBoolean(readFromConfigFile("tdl_require_rec", "true")))
```

Comment:
I feel quite nervous having the `tdl_require_rec` switch in this file.  It is easy to spot and change.

I am thinking to hide the mapping between the config and value in the CredentialConfigFile abstraction.
Here is how this could look:

```
QueueBasedImplementationRunner runner = new QueueBasedImplementationRunner.Builder(implementationRunnerConfig())
                .withSolutionFor("sum", p -> Sum.sum(asInt(p[0]), asInt(p[1])))
                .withSolutionFor("hello", p -> Hello.hello(p[0]))
                .withSolutionFor("fizz_buzz", p -> FizzBuzz.fizzBuzz(asInt(p[0])))
                .withSolutionFor("checkout", p -> Checkout.checkout(p[0]))
                .create();

        ChallengeSession.forRunner(runner)
                .withConfig(challengeSessionConfig())
                .withActionProvider(new UserInputAction(args))
                .start();
```

Reply:
Yeah this is a good idea.